q2_diversity/_beta/_pipeline.py

```diff
@@ -0,0 +1,68 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from q2_diversity_lib.beta import METRICS
+from q2_diversity_lib import translate_metric_name
+
+all_phylo_metrics = METRICS['PHYLO']['IMPL'] | METRICS['PHYLO']['UNIMPL']
+all_nonphylo_metrics = METRICS['NONPHYLO']['IMPL'] \
+                       | METRICS['NONPHYLO']['UNIMPL']
+metric_name_translations = METRICS['METRIC_NAME_TRANSLATIONS']
+
+
+def all_metrics():
+    return all_phylo_metrics | all_nonphylo_metrics
+
+
+def beta_phylogenetic(ctx, table, phylogeny,
+                      metric, threads=1,
+                      variance_adjusted=False,
+                      alpha=None,
+                      bypass_tips=False):
+    if metric not in all_phylo_metrics:
+        raise ValueError("Unknown metric: %s" % metric)
+
+    if alpha is not None and metric != 'generalized_unifrac':
+        raise ValueError('The alpha parameter is only allowed when the choice'
+                         ' of metric is generalized_unifrac')
```

Comment:
Isn't this already implemented in q2-div-lib? Why duplicate it here?

Reply:
This ensures that if users ask `q2_diversity.beta_phylogenetic` to call `weighted_unifrac` or `unweighted_unifrac` with an `alpha`, they are told not to do so. Without this seemingly duplicate guard, the `alpha` argument is silently dropped during dispatch, and the user gets no feedback that they're doing something wrong. I don't like the extra guard, but I like the alternatives less.

Options:
- Duplicate the guard
- User gets no feedback about their error if using `weighted_` or `unweighted_unifrac` with an `alpha`
- We remove the guard from `div-lib` in favor of `diversity.beta_phylogenetic`. This leaves users of `div-lib.beta_phylogenetic_passthrough` with no feedback about their error. Only an improvement because few people will use that directly.
- We add an irrelevant `alpha` param to `q2_diversity_lib.unweighted_unifrac` and `q2_diversity_lib.weighted_unifrac`, for the sole purpose of guarding against this behavior only in `div-lib`. Just no.

I got the whiteboard out for this one, but I may still have missed a better solution. Let me know if you have a better insight.

Reply:
Let's leave as is (duplicate the guard). I added TODOs to both plugins to drop when we can handle optional type-mapped params.